clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -378,15 +378,16 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         memref.instr.type == TRACE_TYPE_PREFETCH_INSTR ||
         memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH) {
         bool expect_encoding = TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_);
-        std::unique_ptr<instr_t> cur_instr_decoded = nullptr;
+        std::unique_ptr<instr_autoclean_t> cur_instr_decoded(
+            new instr_autoclean_t(GLOBAL_DCONTEXT, nullptr));
```

Comment:
This can remain nullptr like before, and assign a new instr_autoclean_t when needed.

Reply:
Done.